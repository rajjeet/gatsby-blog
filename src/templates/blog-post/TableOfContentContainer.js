import React from 'react';
import styled from 'styled-components';
import * as theme from '../../utils/theme';

const TableOfContentContainer = ({ className, htmlContent }) => (
  <div className={className}>
    <h4>Outline</h4>
    <div dangerouslySetInnerHTML={{ __html: htmlContent }} />
  </div>
);

const StyledTableOfContentContainer = styled(TableOfContentContainer)`
    a {
      color: #444;
      :hover {
        color: ${theme.primaryColor};
        cursor: pointer;
      }
    } 
       
    > div > ul {
      list-style-type: none;
      padding-left: 0;
      > li {
        margin-bottom: .3em;
        > p {
        margin-bottom: 0;
        }
      }              
    }
`;

export default StyledTableOfContentContainer;
