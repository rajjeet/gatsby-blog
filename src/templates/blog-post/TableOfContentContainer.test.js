import React from 'react';
import { cleanup, render } from '@testing-library/react';
import TableOfContentContainer from './TableOfContentContainer';

afterEach(cleanup);

describe('BlogPost', () => {
  it('should render', () => {
    const { asFragment } = render(<TableOfContentContainer htmlContent={'<h1>test</h1>'} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display the html markup', () => {
    const { getByText } = render(<TableOfContentContainer htmlContent={'<h1>This blog</h1>'} />);
    expect(getByText('This blog')).toBeDefined();
  });
});
