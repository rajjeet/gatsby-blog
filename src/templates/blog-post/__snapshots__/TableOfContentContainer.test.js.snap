// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa fGalvb"
    id="static-toc"
  >
    <div>
      <h1>
        test
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
